sycl/unittests/scheduler/HostTaskAndBarrier.cpp

```diff
@@ -0,0 +1,288 @@
+//==------------ HostTaskAndBarrier.cpp --- Scheduler unit tests------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "SchedulerTest.hpp"
+#include "SchedulerTestUtils.hpp"
+
+#include <helpers/PiMock.hpp>
+#include <helpers/ScopedEnvVar.hpp>
+#include <helpers/TestKernel.hpp>
+
+#include <vector>
+
+namespace {
+using namespace sycl;
+using EventImplPtr = std::shared_ptr<sycl::detail::event_impl>;
+using ContextImplPtr = std::shared_ptr<sycl::detail::context_impl>;
+using DeviceImplPtr = std::shared_ptr<sycl::detail::device_impl>;
+
+constexpr auto DisableCleanupName = "SYCL_DISABLE_EXECUTION_GRAPH_CLEANUP";
+
+bool CheckTestExecutionRequirements(const sycl::platform &plt) {
+  if (plt.is_host()) {
+    std::cout << "Not run due to host-only environment\n";
+    return false;
+  }
```

Comment:
Host device is dropped so this branch won't be taken.

Reply:
removed https://github.com/intel/llvm/pull/13094/commits/cfb2ef092125226ae5f61e8a03e397a9fab0c298